in_toto/models/statement.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+
+# Copyright New York University and the in-toto contributors
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+<Program Name>
+  statement.py
+
+<Author>
+  Chasen Bettinger <bettingerchasen@gmail.com>
+
+<Started>
+  Mar 25, 2023
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Provides a class that represents a Statement, the middle layer of an 
+  attestation, binding it to a particular subject and unambiguously 
+  identifying the types of the Predicate as defined in the in-toto 
+  attestion spec.
+"""
+
+import attr
+import securesystemslib.formats
+from in_toto.models.common import Signable
+
+STATEMENT_TYPE = "https://in-toto.io/Statement/v1"
+FILENAME_FORMAT = "{step_name}.{keyid:.8}.link"
+FILENAME_FORMAT_SHORT = "{step_name}.link"
+UNFINISHED_FILENAME_FORMAT = ".{step_name}.{keyid:.8}.link-unfinished"
+UNFINISHED_FILENAME_FORMAT_GLOB = ".{step_name}.{pattern}.link-unfinished"
+
+
+@attr.s(repr=False, init=False)
+class Statement(Signable):
+  """The Statement is the middle layer of the attestation, binding it to a 
+  particular subject and unambiguously identifying the types of the Predicate.
+
+  A Link object is usually contained in a generic Metablock object for signing,
+  serialization and I/O capabilities.
+
+  Attributes:
+    subject: A dictionary of software artifacts that the attestation applies to. 
+        Each element represents a single software artifact.
+
+        IMPORTANT: Subject artifacts are matched purely by digest, regardless of 
+        content type. If this matters to you, please comment on GitHub Issue #28
+        (https://github.com/in-toto/attestation/issues/28)
+
+        i.e.::
+
+        {
+            "name": "<NAME>",
+            "digest": {
+                "<ALGORITHM>": "<HEX_VALUE>"
+            }
+        }
+
+    predicate_type: URI identifying the type of the Predicate.
+    
+    predicate: An opaque dictionary containing additional parameters of 
+    the Predicate. Unset is treated the same as set-but-empty. May be 
+    omitted if predicate_type fully describes the predicate.
+  """
+  _type = attr.ib()
+  subject = attr.ib()
+  predicate_type = attr.ib()
+  predicate = attr.ib()
+  MODEL_NAME="Statement"
+
+
+  def __init__(self, **kwargs):
+    super(Statement, self).__init__()
+
+    self._type = STATEMENT_TYPE
+    self.subject = kwargs.get("subject")
+    self.predicate_type = kwargs.get("predicate_type")
```

Comment:
The spec uses camelcasing rather than snake casing. We'd probably be reading in `predicateType` rather than `predicate_type`. Is there any automatic conversion I'm unaware of?